docs/streaming_endpoints.md

```diff
@@ -0,0 +1,50 @@
+
+## Introduction
+OpenVINO Model Server implements gRPC KServe extension which adds separate RPC for [bidirectional streaming](https://grpc.io/docs/what-is-grpc/core-concepts/#bidirectional-streaming-rpc) use cases. It means that besides unary RPC, where client sends a single request and gets back a single response, client is able to initiate connection and send/receive messages in any number and order using `ModelStreamInfer` procedure:
+
+```
+service GRPCInferenceService
+{
+  ...
+  // Standard
+  rpc ModelInfer(ModelInferRequest) returns (ModelInferResponse) {}
+  // Extension
+  rpc ModelStreamInfer(stream ModelInferRequest) returns (stream ModelStreamInferResponse) {}
+  ...
+}
+```
+
+This becomes very useful for endpoints serving [MediaPipe Graphs](./mediapipe.md). In unary inference RPC each request initializes new MediaPipe graph, inputs are deserialized into MediaPipe packets and pushed into the graph. After the graph is done, it is deleted.
+However, in streaming inference RPC MediaPipe graph is created only once per connection, and is reused by subsequent requests to the same gRPC stream. This removes graph initialization overhead and increases its overall throughput.
+
+![diagram](streaming_diagram.svg)
+
+## Graph Selection
+Opening stream is not enough to create execution graph. Instead, the first gRPC request after the opening defines which graph definition will be selected for execution (`model_name` and `model_version` proto fields).
+Afterwards, subsequent requests are required to match the servable name and version, otherwise the error is reported and input packets are not pushed to the graph. However, the graph remains in available for correct requests.
+
+> NOTE: The server closes the stream after the first request if requested graph is non-existent or retired.
+
+## Timestamping
+MediaPipe Graphs require packets to include timestamp information for synchronization purposes. Each input stream in the graph requires timestamps to be monotonically increasing. Read further about [MediaPipe timestamping](https://developers.google.com/mediapipe/framework/framework_concepts/synchronization#timestamp_synchronization).
+
+### Automatic timestamping
+By default OpenVINO Model Server assigns timestamps automatically. Each gRPC request is treated as separate point on the timeline, starting from 0. Each request is deserialized sequentially and increases the timestamp by 1.
+
+> NOTE: It means that in order to send multiple inputs with the same timestamp, client needs to pack it into single request (or use manual timestamping).
+
+### Manual timestamping
+Optionally, the client is allowed to include timestamp manually via request parameter `OVMS_MP_TIMESTAMP`. It is applied to all the packets deserialized from the request.
+
+It is possible to mix manual/automatic timestamping. After correct deserialization step, default automatic timestamp is always equal to `previous_timestamp + 1`.
+
```

Comment:
how the timestamp can be read from the response?

Reply:
Shouldn't it be included in the client snippet section? @bstrzele 
https://github.com/openvinotoolkit/model_server/blob/main/docs/clients_kfs.md

Reply:
it should be in the snippets but we could mention that this value is also returned

Reply:
Added